modules/system/console/OctoberUtil.php

```diff
@@ -339,7 +340,33 @@ protected function utilPurgeOrphans()
             return;
         }
 
-        // @todo
+        $isLocalStorage = Config::get('cms.storage.uploads.disk', 'local') === 'local';
+        $orphanedFiles = FileModel::whereNull('attachment_id')->orWhereNull('attachment_type')->delete();
+
+        foreach (FileModel::all() as $file) {
```

Comment:
Why not just a single `FileModel::whereNotHas('attachment')` instead of loading each file model's attachment?

Reply:
Because this is a `morphTo` relation and I have to specify types for the query:
```
FileModel::whereDoesntHaveMorph('attachment', $types);
```
Beside, I need to be able to check if the file exists on disk when the `--missing-files` argument is passed.

Reply:
Ok, nevermind the "types" argument, I can just pass '*' for types and it works for any model.

2nd argument holds, though.

Reply:
Ok, what about this?